nomad/deploymentwatcher/deployments_watcher.go

```diff
@@ -124,14 +100,22 @@ func NewDeploymentsWatcher(logger *log.Logger, watchers DeploymentStateWatchers,
 }
 
 // SetEnabled is used to control if the watcher is enabled. The watcher
-// should only be enabled on the active leader.
-func (w *Watcher) SetEnabled(enabled bool) error {
+// should only be enabled on the active leader. When being enabled the state is
+// passsed in as it is no longer valid once a leader election has taken place.
+func (w *Watcher) SetEnabled(enabled bool, state *state.StateStore) error {
 	w.l.Lock()
 	defer w.l.Unlock()
 
 	wasEnabled := w.enabled
 	w.enabled = enabled
 
+	if state != nil {
+		if wasEnabled && w.state != nil {
+			panic("we may have a blocking")
```

Comment:
What is this check for? Not sure I get the reason to panic here.

Reply:
Fixed! Was from testing